src/main/java/io/confluent/examples/streams/microservices/PostOrdersAndPayments.java

```diff
@@ -138,29 +138,33 @@ public static void main(final String[] args) throws Exception {
 
             // POST order to OrdersService
             System.out.printf("Posting order to: %s   .... ", path.urlPost());
-            final Response response = client.target(path.urlPost())
-                                            .request(APPLICATION_JSON_TYPE)
-                                            .post(Entity.json(inputOrder));
-            System.out.printf("Response: %s %n", response.getStatus());
-
-            // GET the bean back explicitly
-            System.out.printf("Getting order from: %s   .... ", path.urlGet(i));
-            returnedOrder = client.target(path.urlGet(i))
-                                  .queryParam("timeout", Duration.ofMinutes(1).toMillis() / 2)
-                                  .request(APPLICATION_JSON_TYPE)
-                                  .get(newBean());
-            if (!inputOrder.equals(returnedOrder)) {
-                System.out.printf("Posted order %d does not equal returned order: %s%n", i, returnedOrder.toString());
-            } else {
-                System.out.printf("Posted order %d equals returned order: %s%n", i, returnedOrder.toString());
+            try {
+                final Response response = client.target(path.urlPost())
+                        .request(APPLICATION_JSON_TYPE)
+                        .post(Entity.json(inputOrder));
+                System.out.printf("Response: %s %n", response.getStatus());
+
+                // GET the bean back explicitly
+                System.out.printf("Getting order from: %s   .... ", path.urlGet(i));
+                returnedOrder = client.target(path.urlGet(i))
+                        .queryParam("timeout", Duration.ofMinutes(1).toMillis() / 2)
+                        .request(APPLICATION_JSON_TYPE)
+                        .get(newBean());
+                if (!inputOrder.equals(returnedOrder)) {
+                    System.out.printf("Posted order %d does not equal returned order: %s%n", i, returnedOrder.toString());
+                } else {
+                    System.out.printf("Posted order %d equals returned order: %s%n", i, returnedOrder.toString());
+                }
+
+                // Send payment
+                final Payment payment = new Payment("Payment:1234", id(i), "CZK", 1000.00d);
```

Comment:
Is it ok to send the order above and fail sending the payment? Or should sending the payment have it's own try/catch block?

Reply:
Thanks Mike, I took the fastest route I could come up with.  The thing I was trying to solve was the order service http service not yet available, which is the common issue for the demos we use this in.  That's the first thing in this block so my assumption was it would fail there and then retry the whole block again.